pandas/core/sparse/array.py

```diff
@@ -195,7 +195,7 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
                 data = np.nan
             if not is_scalar(data):
                 raise Exception("must only pass scalars with an index ")
-            values = np.empty(len(index), dtype='float64')
+            values = np.empty(len(index), dtype=infer_dtype_from(data)[0])
```

Comment:
Is `infer_dtype_from_scalar` more appropriate here, since we've validated that `data` is a scalar?

Reply:
yea that's a good point. Since infer_dtype_from just checks is_scalar again.